pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1069,7 +1070,13 @@ func buildDestination(ctx RouteContext, to k8s.BackendRef, ns string,
 			invalidBackendErr = &ConfigError{Reason: InvalidDestinationNotFound, Message: fmt.Sprintf("backend(%s) not found", hostname)}
 		}
 	case gvk.InferencePool:
-		if !features.SupportGatewayAPIInferenceExtension || strings.Contains(string(to.Name), ".") {
+		if !features.SupportGatewayAPIInferenceExtension {
+			return nil, nil, &ConfigError{
+				Reason:  InvalidDestinationKind,
+				Message: "InferencePool is not supported in this version of Istio",
```

Comment:
should this be "InferencePool is not supported, to enable do X"?

Reply:
I don't think we typically add that sort of thing in status (e.g. if alpha gateway api isn't enabled, we don't write anything at all). If you/others feel strongly though, I'm happy to add it in

Reply:
its just that I think this is misguiding the user. Not supported in this version does not tell me if I have any way to support it.

Reply:
Fair enough; edited